### PR TITLE
feat: additional dubbd helm values via zarf var

### PR DESCRIPTION
**Pull request by MxNxPx**
to address https://github.com/defenseunicorns/uds-package-dubbd/issues/422

limitation to this is that it only allows the user to modify dubbd package helm chart values but does not support adding new ones (since this is injected at package deploy time, not package create time). also, the bigbang extension builds out the full zarf yaml at package create time, so toggling off a helm chart via `enabled: false` does not work as the zarf.yaml is expecting it and wait to for the package to be ready (eventually timeout/fail).

### CODE REVIEW

File: aws/dubbd-aws/zarf.yaml [hunk 1]
```diff
@@ -49,6 +49,13 @@ variables:
     description: "List of namespaces to be exempt from cluster policies (must contain zarf)"
     default: '[ "zarf", "calico-apiserver", "calico-system", "tigera-operator" ]'
 
+  - name: ADD_DUBBD_VALUES
+    default: deploy-add-dubbd-values.yaml
+    prompt: false
+    type: file
+    autoIndent: true
```
Comment (mjnagel) [inline aws/dubbd-aws/zarf.yaml:56]: I'm not so convinced anymore on the value of having a separate "extra values" specific to AWS/k3d/rke2...if someone is deploying DUBBD AWS what is the benefit of having two spots to override values? They could leverage the one from the base DUBBD, even if deploying AWS and I don't think there's any extra benefit to having two values files?

File: defense-unicorns-distro/zarf.yaml [hunk 2]
```diff
@@ -54,6 +54,11 @@ variables:
   - name: EPHEMERAL
     description: "Set whether the cluster should be considered ephemeral - if true all resources will be force destroyed on removal"
     default: "false"
+  - name: DUBBD_VALUES
+    default: deploy-dubbd-values.yaml
```
Comment (mjnagel) [inline defense-unicorns-distro/zarf.yaml:58]: I don't have a great suggestion here but I wonder if we can find a default that won't throw a warning on deploy if it's not provided 🤔 

```
 WARNING  Refusing to load a non-text file for templating ###ZARF_VAR_DUBBD_VALUES###                                                                                                                                   
                                                                                                                                                                                                                        
 WARNING  Refusing to load a non-text file for templating ###ZARF_VAR_ADD_DUBBD_VALUES###                                                                                                                               
```

**Review by mjnagel — COMMENTED** [review]
Overall I like it. Do you think we need docs on intended usage and what is/isn't supported? Thinking primarily of someone stumbling on this and trying to enable/disable a package which wouldn't work. *Maybe* also something to encourage less usage of this and more usage of issues to suggest default values for DUBBD?

I know we don't necessarily want this "advertised" for usage, but I think it would be good to document it with caution signs and bold warnings to keep us out of trouble 😆

File: aws/dubbd-aws/zarf.yaml (hunk 1 above)
Comment (mjnagel) [reply aws/dubbd-aws/zarf.yaml:56]: As discussed synchronously - this was added to allow overriding the specific values for AWS, k3d, etc (due to the values file ordering). Might be possible to share the same zarf var to accomplish this same purpose...

File: defense-unicorns-distro/zarf.yaml (hunk 2 above)
Comment (MxNxPx) [reply defense-unicorns-distro/zarf.yaml:59]: @mjnagel - good catch!
addressed by including a default file (like [this one](https://github.com/defenseunicorns/uds-package-dubbd/blob/dubbd-values/defense-unicorns-distro/deploy-dubbd-values.yaml)) in the repo for each dubbd package and no longer receive that warning

**Review by mjnagel — APPROVED** [review]
:shipit:

File: aws/dubbd-aws/zarf.yaml (hunk 1 above)
Comment (MxNxPx) [reply aws/dubbd-aws/zarf.yaml]: @mjnagel - updated to use the same zarf var regardless of the package which will write to both the dubbd base values override file and the dubbd values override file for each package ([dubbd-aws](https://github.com/defenseunicorns/uds-package-dubbd/blob/dubbd-values/aws/dubbd-aws/zarf.yaml#L52), [rke2](https://github.com/defenseunicorns/uds-package-dubbd/blob/dubbd-values/rke2/zarf.yaml#L29), [k3d](https://github.com/defenseunicorns/uds-package-dubbd/blob/dubbd-values/k3d/zarf.yaml#L26)).  although redundant, it doesn't harm anything, and it simplifies it to a single zarf var that has the same impact of being the last values file with the highest precedence.

**Comment by MxNxPx** [comment]
> docs on intended usage and what is/isn't supported? Thinking primarily of someone stumbling on this and trying to enable/disable a package which wouldn't work (or switch

added a description in the zarf.yaml files

**Pull request merged**